test/extensions/stats_sinks/metrics_service/metrics_service_integration_test.cc

```diff
@@ -94,7 +94,8 @@ class MetricsServiceIntegrationTest : public HttpIntegrationTest,
     }
     EXPECT_TRUE(known_counter_exists);
     EXPECT_TRUE(known_gauge_exists);
-    EXPECT_TRUE(known_histogram_exists);
+    // TODO : Fix the timing issue with stats flush - This is flaky.
```

Comment:
This check is flaky. Some times it fails because of timing issue as mentioned in https://github.com/envoyproxy/envoy/issues/3238. Not just in Valgrind, it fails some times in Mac also. Looks like it is flaky. Will look in to this.

Reply:
I would think there should be some way to ensure that a sample was recorded and that a flush was called strictly after that, but I'm not familiar with the internals of this test.

Reply:
I need to look in to in more detail. I am suspecting some times (though very rarely) initial flush does not have this histogram. While the flakiness is very rare (I have observed only once), I am little concerned that other PRs might be affected by this. So I am commenting it for now. BTW, I am travelling today and over the weekend, mostly busy next week with meetings, will look in to this as soon as I can. @mattklein123 WDYT?  Let me know if you have any better ideas.

Reply:
It would be very good to fix this.  Flaky tests are a drag on developer velocity.  OTOH commenting out the test means that the although the test-coverage may show look OK, you really aren't testing the code :)

Any chance you can dive in and debug as part of this PR?

Reply:
As I mentioned, I can not do it immediately as I am travelling and early part of next week as well. If it is OK to leave this PR open, I can dig deep but I am little concerned about other PRs. If you have any quick ideas, do let me know, I can try quickly.

Reply:
The thing that comes to mind is to just wait on the histogram existing, vs. it existing after a single metrics request. I.e., allow there to be more than 1 request until you find the histogram. This will paper of the timing issue if we don't have it in time for the first flush.